realm/realm-library/src/objectServer/java/io/realm/RealmAppConfiguration.java

```diff
@@ -175,10 +184,25 @@ public File getSyncRootDirectory() {
         return syncRootDir;
     }
 
+    // FIXME Doc
+    public CodecRegistry getDefaultCodecRegistry() { return codecRegistry; }
+
     /**
      * FIXME
      */
     public static class Builder {
+        // Default BSON codec for passing BSON to/from JNI
+        static CodecRegistry DEFAULT_BSON_CODEC_REGISTRY = CodecRegistries.fromRegistries(
```

Comment:
Just stumbled upon the `BsonUtils.java` class in the Stitch SDK. The use this:
```
  public static final CodecRegistry DEFAULT_CODEC_REGISTRY =
      fromProviders(
          asList(
              new ValueCodecProvider(),
              new BsonValueCodecProvider(),
              new DocumentCodecProvider(),
              new IterableCodecProvider(),
              new MapCodecProvider()));
```

Reply:
@rorbech Any thoughts on the above ☝️ ? I would assume we would go with the broadest possible set of defaults, so whatever type people bring there is a fair good chance it will work?

Reply:
Did not evaluate all providers/codecs yet, but just added what I immediately spotted as userful. 
The `MapCodecProvider` sounds like an obvious candidate, and from @edualonso it sounded like the `DocumentCodecProvider` would be useful for the collections. 
So the above `DEFAULT_CODEC_REGISTRY` sound like a good update, but given the parser issues lower down the stack I would prefer awaiting usage/tests and add them in another PR to avoid pushing this one further.

Reply:
👍

Reply:
I added it as TODO on #6810